### ISSUE TITLE
API ref: Expose docstring on class attributes

### ISSUE THREAD

**Issue by jeremyliweishih** [alteryx/evalml#652]
Currently we showcase the value of class attributes when available (see API reference for specific pipelines). However this doesn't show the docstring of these attributes and may be confusing since these attributes may rely other fields (such as `custom_name` and `name`).

We should also include another table similar to instance attributes but for class attributes.

**Comment by freddyaboulton** [comment]
I took a stab at this and am having trouble using autodoc for two reasons:

1. The class properties of pipeline base classes depend on the component graph - which isn't defined when we generate the docs. Let's say we tell autodoc to generate documentation for the `hyperparameters` classproperty for `ClassificationPipeline`. Under the hood, it will import our class, and then call `ClassificationPipeline.hyperparameters`. Because of how `classproperty` works, this will try to compute the hyperparameters by iterating over the component graph but since the component graph is not set, we get an error. 

2. When the class property does not depend on an abstract attribute, it is evaluated and autodoc won't access the docstring. For example, `CatBoostClassifier.needs_fitting` evaluates to True. If we were to call `.__doc__` on this, we'd get the docstring for the built-in bool type. We can't overwrite the docstrings for built-in types. This is what @jeremyliweishih saw when he called `.__doc__` on the pipeline name above.

autodoc is able to add docstrings for properties because when you call `ClassName.property_name` you get back an instance of `property` which has a `__doc__` attribute (same with methods and functions).

In short, I think this is hard to do with the built-in sphinx autodoc tools given how `classproperty` is intended to work.

A possible workaround is to write our own sphinx extension for extracting the docstrings from source code but that sounds like a heavy lift.

I am not a sphinx expert so let me know what you guys think!
